baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/SafeLoggingPropagationTest.java

```diff
@@ -249,13 +249,284 @@ void testIgnoresStaticMethods() {
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import org.immutables.value.Value;",
+                        "@Value.Immutable",
+                        "abstract class Test {",
+                        "  @DoNotLog",
+                        "  String token() { return \"\"; }",
+                        "}")
+                .expectUnchanged()
+                .doTest();
+    }
+
+    @Test
+    void testIncludesJacksonAnnotatedHelperMethods_iface() {
+        fix().addInputLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import com.fasterxml.jackson.annotation.*;",
+                        "import org.immutables.value.Value;",
+                        "@Value.Immutable",
+                        "interface Test {",
+                        "  @DoNotLog",
+                        "  @JsonProperty",
+                        "  default String token() { return \"\"; }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import com.fasterxml.jackson.annotation.*;",
+                        "import org.immutables.value.Value;",
+                        "@DoNotLog",
+                        "@Value.Immutable",
+                        "interface Test {",
+                        "  @DoNotLog",
+                        "  @JsonProperty",
+                        "  default String token() { return \"\"; }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testIncludesJacksonAnnotatedHelperMethods_abstract() {
+        fix().addInputLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import com.fasterxml.jackson.annotation.*;",
+                        "import org.immutables.value.Value;",
+                        "@Value.Immutable",
+                        "abstract class Test {",
+                        "  @DoNotLog",
+                        "  @JsonProperty",
+                        "  String token() { return \"\"; }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import com.fasterxml.jackson.annotation.*;",
+                        "import org.immutables.value.Value;",
+                        "@DoNotLog",
+                        "@Value.Immutable",
+                        "abstract class Test {",
+                        "  @DoNotLog",
+                        "  @JsonProperty",
+                        "  String token() { return \"\"; }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testIgnoresJsonIgnoreHelperMethod() throws Exception {
```

Comment:
this test would have been an ignore even without the JsonIgnore annotation. Is it worth a version of this test that would actually propagate the method safety annotaiton to the class if the JsonIgnore wasn't there?

Reply:
I'll add another test with a jsonignored field, however we _do_ want to include those in safety because they're included in the toString.